src/listener.rs

```diff
@@ -36,11 +39,25 @@ impl TfoListener {
         #[cfg(not(windows))]
         socket.set_reuseaddr(true)?;
 
+        // On all other platforms, TCP_FASTOPEN can be set before bind(), between bind() and listen(), or after listen().
+        // We prefer setting it before bind() as this feels most like the natural order of socket initialization sequence.
+        //
+        // On macOS, setting TCP_FASTOPEN_FORCE_ENABLE requires the socket to be in the TCPS_CLOSED state.
+        // TCP_FASTOPEN, on the other hand, can only be set when the socket is in the TCPS_LISTEN state.
+        cfg_if! {
+            if #[cfg(not(target_os = "macos"))] {
```

Comment:
I guess you may also want to enable it for iOS, watchOS, ...

Reply:
Oops, I got too used to Go's `darwin` and forgot that in Rust we need to distinguish between these OSes. 😅